File: AlkmaarseRestaurants/angular/src/app/shared/Models/RestaurantModel.ts
import { IRestaurantModel } from "./IRestaurantModel";
import { RestaurantInfoModel } from "./RestaurantInfoModel";
import { RestaurantMenuItemModel } from "./RestaurantMenuItemModel";

export class RestaurantModel implements IRestaurantModel {
    public id: number;
    public name: string;
    public address: string;
    public streetName: string;
    public streetNumber: string;
    public imageLogo: string;
    public restaurantType: string;
    public restaurantInfo: RestaurantInfoModel;
    public restaurantMenu: Array<RestaurantMenuItemModel>;

    constructor() {
        this.id = 0;
        this.name = "";
        this.address = "";
        this.streetName = "";
        this.streetNumber = "";
        this.imageLogo = "";
        this.restaurantType = "";
        this.restaurantInfo = new RestaurantInfoModel();
        this.restaurantMenu = [];
    }
}

File: AlkmaarseRestaurants/Models/RestaurantModel.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace AlkmaarseRestaurants.Models
{
    public class RestaurantModel
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Address { get; set; }
        public string StreetName { get; set; }
        public int StreetNumber { get; set; }
        public string ImageLogo { get; set; }
        public string RestaurantType { get; set; }
    }
}
